test/test_prototype_transforms.py

```diff
@@ -152,3 +153,56 @@ def test_normalize(self, transform, input):
     )
     def test_random_resized_crop(self, transform, input):
         transform(input)
+
+
+class TestRandomHorizontalFlip:
+    def input_tensor(self, dtype=torch.float32):
+        return torch.tensor([[[0, 1], [0, 1]], [[1, 0], [1, 0]]], dtype=dtype)
+
+    def expected_tensor(self, dtype=torch.float32):
+        return torch.tensor([[[1, 0], [1, 0]], [[0, 1], [0, 1]]], dtype=dtype)
+
+    @pytest.mark.parametrize("p", [0.0, 1.0], ids=["p=0", "p=1"])
+    def test_simple_tensor(self, p):
+        input = self.input_tensor()
+
+        actual = transforms.RandomHorizontalFlip(p=p)(input)
+
+        expected = self.expected_tensor() if p == 1.0 else input
+        assert_equal(expected, actual)
+
+    @pytest.mark.parametrize("p", [0.0, 1.0], ids=["p=0", "p=1"])
+    def test_pil_image(self, p):
+        input = self.input_tensor(dtype=torch.uint8)
+
+        actual = transforms.RandomHorizontalFlip(p=p)(to_pil_image(input))
+
+        expected = self.expected_tensor(dtype=torch.uint8) if p == 1.0 else input
+        assert_equal(expected, pil_to_tensor(actual))
+
+    @pytest.mark.parametrize("p", [0.0, 1.0], ids=["p=0", "p=1"])
+    def test_features_image(self, p):
+        input = self.input_tensor()
+
+        actual = transforms.RandomHorizontalFlip(p=p)(features.Image(input))
+
+        expected = self.expected_tensor() if p == 1.0 else input
+        assert_equal(features.Image(expected), actual)
+
+    @pytest.mark.parametrize("p", [0.0, 1.0], ids=["p=0", "p=1"])
+    def test_features_segmentation_mask(self, p):
+        input = features.SegmentationMask(self.input_tensor())
+
+        actual = transforms.RandomHorizontalFlip(p=p)(input)
+
+        expected = self.expected_tensor() if p == 1.0 else input
+        assert_equal(features.SegmentationMask(expected), actual)
+
+    @pytest.mark.parametrize("p", [0.0, 1.0], ids=["p=0", "p=1"])
+    def test_features_bounding_box(self, p):
+        input = features.BoundingBox([0, 0, 5, 5], format=features.BoundingBoxFormat.XYXY, image_size=(10, 10))
+
+        actual = transforms.RandomHorizontalFlip(p=p)(input)
+
+        expected = torch.tensor([5, 0, 10, 5]) if p == 1.0 else input
+        assert_equal(features.BoundingBox.new_like(input, expected), actual)
```

Comment:
Could you also check that the meta data of the feature stays the same?

```suggestion
        assert_equal(features.BoundingBox.new_like(input, expected), actual)
        assert actual.format == expected.format
        assert actual.image_size == expected.image_size
```

? 

I'm currently working on adding such functionality to assert equal, so we don't have to do it manually. But for now we should make sure here.